src/app/components/service/service.component.ts

```diff
@@ -221,4 +246,65 @@ export class ServiceComponent implements OnInit, OnDestroy {
         return 'g_translated'
     }
   }
+
+  updateMessage(data: EmittedData, id: string): void {
+    const { event, index } = data
+    const value = (event.target as HTMLTextAreaElement).innerText
+
+    const translation = this.translations$()[index].clone()
+    translation.messages = this.translations$()[index].messages.filter((message) => {
```

Comment:
In functional programming, side effects are bad.

In this case, you change the value of `this.state` and `message.message` in `filter`. You MUST never have side effects in `filter`.

Can you write it so there are no side effects?